csrc/scheduler/reduction_utils.cpp

```diff
@@ -556,7 +556,8 @@ void propagateParallelization(
         std::back_inserter(allreduce_tvs),
         [&](auto tv) {
           return reduction_tv != tv &&
-              reduction_scheduler_utils::isGridAllreduce(tv);
+              (reduction_scheduler_utils::isGridAllreduce(tv) ||
```

Comment:
Naive question, could you elaborate why we should propagate the ParallelType::Group to `isGridAllreduce` tensors?

Reply:
Looking at the code, I think we don't need this `isGridAllreduce` check. I don't know why it was there, maybe it was used because we will use grouped reduction if the fusion has GridAllReduce. I deleted it and let's see how CI goes.

Reply:
I think the reason was grouping was only supported in certain cases. Looks like the condition of applying grouping is now changed to follow the `use_grouped_reduction`. What does this change mean? Is there any case where previously grouping was used but not after this PR?

Reply:
Previously (before adding iter grouped outer reducion), the grouped reduction is only allowed for outer-grid-persistent kernels. After adding `iter grouped outer reducion` we extends to non-persistent outer grid/block reduction. There is no change where `previously grouping was used but not after this PR`.

I think these code assumes that for fusion  with `GridAllReduce`  Tvs, we will always do grouped reduciton, which is true as in the current `schedulePersistentKernel`, we have:
```
  const bool is_outer_grid_persistence = rparams.persistent_kernel &&
      rparams.cross_grid_inner_reduction && !rparams.fastest_dim;
  reduction_scheduler_utils::multiReductionInliner(
      fusion,
      reduction_tvs[0],
      reference_tv,
      unroll,
      vectorize,
      is_outer_grid_persistence,
      reduction_tvs,
      cached_inputs,
      cached_outputs,
      dummy_outputs);
```
The para `is_outer_grid_persistence` corresponds to `use_grouped_reduction`. Maybe we should rename it.